### DISCUSSION TITLE
Add "children" as a default slot name

### DISCUSSION THREAD

**Discussion by gone** [django-components/django-components#124] [Ideas]
For the component: 

```
<div class="calendar-component">
    <div class="header">
        {% slot "header" %}Calendar header{% endslot %}
    </div>
    <div class="body">
        {% slot "body" %}Today's date is <span>{{ date }}</span>{% endslot %}
    </div>
</div>
```

With usage: 

```
{% component_block "calendar" date="2020-06-06" %}
    {% slot "body" %}Can you belive it's already <span>{{ date }}</span>??{% endslot %}
{% endcomponent_block %}
```

It would be nice to have this declared/used like:

```
<div class="calendar-component">
    <div class="header">
        {% slot "header" %}Calendar header{% endslot %}
    </div>
    <div class="body">
        {% slot "children" %}Today's date is <span>{{ date }}</span>{% endslot %}
    </div>
</div>
```

```
{% component_block "calendar" date="2020-06-06" %}
    Can you belive it's already <span>{{ date }}</span>??
{% endcomponent_block %}
```

I know I put a bunch of different ideas up - let me know if any of them sound like directions you are interested in going and I can put together a pr.

**Comment by EmilStenstrom** [comment]
I appreciate your goal of simplifying the syntax with this suggestion, but I don't think this is the right way. I'm imagining a developer new to django-components, looking at a mature code-base that uses this library. Looking at the example above, why does the content go in the second block, and not the first one? You just have to know that "children" is a special name, that means something different from the other names. I'd like to avoid that kind of library-specific knowledge if possible. Often this means being explicit, and repeating yourself a bit, but I think it's worth it because if acts as documentation for the next developer (which could be myself months later).

Appreciate your suggestions!

**Comment by JuroOravec** [comment]
Closing as this seems resolved.

For anyone who comes across this thread, the current way (May 2024) to define a default slot, is to append a `default` keyword (e.g. `{% slot "my_slot" default %}`), so you can use any name for the slot.
